packages/astro/src/core/config.ts

```diff
@@ -397,7 +397,59 @@ export async function resolveConfigURL(
 	}
 }
 
-/** Attempt to load an `astro.config.mjs` file */
+interface OpenConfigResult {
+	userConfig: AstroUserConfig;
+	astroConfig: AstroConfig;
+	flags: CLIFlags;
```

Comment:
bonus: since you've built it... it would be great to track `Object.keys(flags)` via telemetry as well. I can see this being pretty useful data for designing flags vs. config options.

Reply:
Yep sounds good, think I'll do this one in a followup since I want to write tests.

Reply:
Did it in this PR.